### PR TITLE
Add  syntax highlighting for juvix code blocks in docs

### PR DESCRIPTION
**Pull request by jonaprieto**
This PR adds initial syntax highlighting for juvix code blocks and REPL sessions in Markdown files rendered by mdbook. After this PR, only two themes would be supported to ease maintenance: Light and Dark (Ayu).

The implementation is a specifically tailored version of

- https://github.com/anoma/highlightjs-juvix (plugin for HighlightJS,v11.7).

to be compatible with the infamous HighlightJS 10.1.1, to be used just for MdBook.

The output can be seen here (make sure the CI finished to check the last version, otherwise run the website locally):

- https://jonaprieto.github.io/juvix/tutorials/learn.html

### CODE REVIEW

File: docs/tutorials/learn.md [hunk 1]
```diff
@@ -27,10 +27,10 @@ juvix repl
 
 The response should be similar to:
 
-```juvix
+```jrepl
 Juvix REPL version 0.3: https://juvix.org. Run :help for help
 OK loaded: ./.juvix-build/stdlib/Stdlib/Prelude.juvix
-Stdlib.Prelude>
+Stdlib.Prelude> 1 + 3
```
Comment (lukaszcz) [inline docs/tutorials/learn.md:33]: Why add `1 + 3` here? This is not a response that one typically sees after just launching the REPL.

File: docs/tutorials/learn.md [hunk 2]
```diff
@@ -515,7 +515,7 @@ polymorphic definition of lists from the standard library:
 infixr 5 ::;
 type List (A : Type) :=
 | nil : List A
-| :: : A -> List A -> List A;
+| :: : {A : Type} -> A -> List A -> List A;
```
Comment (lukaszcz) [inline docs/tutorials/learn.md:518]: This is wrong. That's not the definition of `::` from the standard library, and I think Juvix doesn't accept explicitly stating the constructor parameters?

File: docs/tutorials/learn.md [hunk 3]
```diff
@@ -569,7 +569,7 @@ This is not acceptable if you care about performance. In an imperative
 language, one would use a simple loop going over the list without any
 memory allocation. In pseudocode:
 
-```juvix
+```
```
Comment (lukaszcz) [inline docs/tutorials/learn.md:572]: Maybe adding `pascal` or something like this here would give reasonable highlighting? Do we have access to predefined highlighting for other languages?

File: docs/tutorials/learn.md (hunk 1 above)
Comment (jonaprieto) [reply docs/tutorials/learn.md:33]: Thanks, I changed it during the testing phase, same for the feedback back.

File: docs/tutorials/learn.md (hunk 2 above)
Comment (jonaprieto) [reply docs/tutorials/learn.md:518]: Indeed, I wanted to see the colours here, but forgot to delete it.

File: docs/tutorials/learn.md [hunk 4]
```diff
@@ -569,7 +569,7 @@ This is not acceptable if you care about performance. In an imperative
 language, one would use a simple loop going over the list without any
 memory allocation. In pseudocode:
 
-```juvix
+```pascal
```
Comment (lukaszcz) [inline docs/tutorials/learn.md:572]: We can try "delphi" or "Delphi" here. Then some reasonable highlighting should work.

Comment (jonaprieto) [reply docs/tutorials/learn.md:572]: Done.

**Review by lukaszcz — APPROVED** [review]

**Pull request merged**